gps_rviz/src/gps_handler.cpp

```diff
@@ -0,0 +1,122 @@
+#include "geometry_msgs/Vector3.h"
+#include "math.h"
+#include "visualization_msgs/Marker.h"
+#include "visualization_msgs/MarkerArray.h"
+#include "gps_handler.h"
+
+#define PI 3.14159265
+
+using namespace std;
+
+namespace gps{
+
+ geometry_msgs::Vector3 InitializingPoint, InitializingPoint_missionframe;
+ bool REFERENCE_FRAME_IS_INITIALIZED;
+ float alpha_init, beta_init;
+ 
+ visualization_msgs::MarkerArray GPS_TRACK;
+ visualization_msgs::Marker gps_points, gps_line_strip, gps_line_list;
+
+// ---------------------------------------------
+// Rotational Matrix R_0G=R_01*R_1G; R_01=Rz(a); R_1G=Ry(b)
+geometry_msgs::Vector3 transformation(geometry_msgs::Vector3 Input, float a, float b){
+	geometry_msgs::Vector3 Output;
+ 	Output.x = cos(a)*cos(b)*Input.x - sin(a)*Input.y +cos(a)*sin(b)*Input.z;
+ 	Output.y = sin(a)*cos(b)*Input.x + cos(a)*Input.y +sin(a)*sin(b)*Input.z;
+ 	Output.z = -sin(b)*Input.x + cos(b)*Input.z;
+
+return Output;
+}
+
+// -------------------------------------------------
+// Rotational Matrix plus shifting
+geometry_msgs::Vector3 transformToMissionFrame(geometry_msgs::Vector3 InputPoint){
```

Comment:
- Return value as function argument.
- Pass input arguments that do not change as pointers or by const reference.
```
void transformToMissionFrame(const geometry_msgs::Vector3& input_point, geometry_msgs::Vector3& transformed_point) {
```